### PR TITLE
Reschedule service if instance deploy_rev does not match

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

**Comment by jakolehm** [comment]
@jnummelin @nevalla ping

File: server/app/services/grid_service_deployer.rb [hunk 1]
```diff
@@ -60,10 +60,9 @@ def deploy
     info "starting to deploy #{self.grid_service.to_path}"
     ping_subscription = self.subscribe_to_ping
     creds = self.creds_for_registry
-    self.grid_service.set_state('deploying')
-    self.grid_service.set(:deployed_at => Time.now.utc)
-
     deploy_rev = Time.now.utc.to_s
+    self.grid_service.set_state('deploying')
+    self.grid_service.set(:deployed_at => deploy_rev)
```
Comment (SpComb) [inline server/app/services/grid_service_deployer.rb]: This assigns a String to the `GridService.deployed_at` `type: DateTime` attribute?

File: server/app/jobs/service_balancer_job.rb [hunk 2]
```diff
@@ -118,6 +118,7 @@ def pending_deploys?(service)
   def lagging_behind?(service)
     return true if service.deployed_at && service.updated_at > service.deployed_at
     return true if service.deployed_at && service.deploy_requested_at && service.deploy_requested_at > service.deployed_at
+    return true if service.deployed_at && service.containers.where(:deploy_rev.ne => service.deployed_at).count > 0
```
Comment (SpComb) [inline server/app/jobs/service_balancer_job.rb]: This compares the `Container.deploy_rev` String with the `GridServer.deployed_at` DateTime?

**Comment by jnummelin** [comment]
Looks good

File: server/app/jobs/service_balancer_job.rb (hunk 2 above)
Comment (jakolehm) [inline server/app/jobs/service_balancer_job.rb]: Ok, have to agree that something is off here... will investigate / add better tests.

**Comment by jakolehm** [comment]
Let's close this one, with stacks we need to change the logic anyway.

**Pull request closed without merging**